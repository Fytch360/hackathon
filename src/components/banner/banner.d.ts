import { Status } from '../..'

export type BannerDescription = string | React.ReactNode

export type BannerColorScheme = keyof Omit<Status, 'primary'>

export interface BannerProps extends React.HTMLProps<HTMLDivElement> {
    title?: string
    description?: BannerDescription
    children?: React.ReactNode
    colorScheme?: BannerColorScheme
    imageSrc?: string
    srcSet?: string
}

export const Banner: React.FC<BannerProps>
