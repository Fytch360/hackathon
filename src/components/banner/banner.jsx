import React from 'react'
import PropTypes from 'prop-types'

import { Headline3, Typography } from '../typography'
import { PaddingWrapper } from '../indent-wrapper/padding-wrapper.style'

import {
    ContainerStyled,
    ContentStyled,
    ImageWrapperStyled,
    ImgStyled,
} from './banner.style'

export const Banner = ({
    title,
    description,
    children,
    colorScheme,
    imageSrc,
    srcSet,
    className
}) => (
    <ContainerStyled colorScheme={colorScheme} className={className}>
        <ContentStyled size="h3" verticalPadding="inner">
            {title && <Headline3 indent="zero">{title}</Headline3>}

            {description && (
                <Typography
                    as="div"
                    size="md"
                    verticalMargin="micro"
                    colorScheme="secondary"
                >
                    {description}
                </Typography>
            )}

            {children && (
                <PaddingWrapper verticalPadding="inner">
                    {children}
                </PaddingWrapper>
            )}
        </ContentStyled>

        <ImageWrapperStyled>
            <ImgStyled src={imageSrc} srcSet={srcSet} alt={title} />
        </ImageWrapperStyled>
    </ContainerStyled>
)

Banner.propTypes = {
    title: PropTypes.string,
    description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    children: PropTypes.node,
    colorScheme: PropTypes.oneOf(['success', 'warning', 'info', 'draft']),
    imageSrc: PropTypes.string,
    srcSet: PropTypes.string,
    className: PropTypes.string
}
