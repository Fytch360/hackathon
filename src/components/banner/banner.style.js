import styled from "@emotion/styled/macro"
import { css } from '@emotion/react'

import { Container } from '../indent-wrapper/container'
import { smBorderRadius } from '../styles/radius.config.style'
import { smShadow } from '../styles/shadows.config.style'
import { dynamicIndent } from '../styles/dynamic-styles'

const dynamicBackground = (colorScheme, theme) => {
    switch (colorScheme) {
        case 'success':
            return theme.bannerSuccess
        case 'info':
            return theme.bannerInfo
        case 'warning':
            return theme.bannerWarning
        case 'draft':
            return theme.bannerDraft
        default:
            return theme.noColor
    }
}

export const ContainerStyled = styled.div(({ theme, colorScheme }) => css`
    display: flex;
    background-color: ${dynamicBackground(colorScheme, theme)};
    min-height: 156px;
    padding-left: ${dynamicIndent('h3', 'inner')};
    border-radius: ${smBorderRadius};
    box-shadow: ${smShadow(theme)};

    ${theme.media?.sm && css`
        @media ${theme.media.sm} {
            min-height: auto;
            padding-right: ${dynamicIndent('h3', 'inner')};
        }
    `}
`)

export const ContentStyled = styled(Container)(({ theme }) => css`
    width: 344px;
    flex-shrink: 0;

    ${theme.media?.sm && css`
        @media ${theme.media.sm} {
            width: 100%;
        }
    `}
`)

export const ImageWrapperStyled = styled.div(({ theme }) => css`
    position: relative;
    overflow: hidden;
    width: 100%;

    ${theme.media?.sm && css`
        @media ${theme.media.sm} {
            display: none;
        }
    `}
`)

export const ImgStyled = styled.img`
    position: absolute;
    top: 0;
    left: 50%;
    transform: translate(-50%);
    height: 100%;
    width: auto;
    pointer-events: none;
`
