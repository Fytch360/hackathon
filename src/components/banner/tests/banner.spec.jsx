import React from 'react'
import { shallow } from 'enzyme'

import { Banner } from '../banner'

describe('<Banner />', () => {
    it('is available', () => {
        expect(Banner).toBeDefined()
    })


    it('renders title, description, children', () => {
        const wrapper = shallow(
            <Banner title="Заголовок" description="описание">
                <span>{'foo bar baz'}</span>
            </Banner>
        )
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('Headline3').length).toBe(1)
        expect(wrapper.contains('Заголовок')).toBeTruthy()
        expect(wrapper.contains('описание')).toBeTruthy()
        expect(wrapper.contains(<span>{'foo bar baz'}</span>)).toBeTruthy()
    })
})
