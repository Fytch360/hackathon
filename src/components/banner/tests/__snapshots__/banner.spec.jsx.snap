// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders title, description, children 1`] = `
<ContainerStyled>
  <ContentStyled
    size="h3"
    verticalPadding="inner"
  >
    <Headline3
      indent="zero"
    >
      Заголовок
    </Headline3>
    <TypographyStyled
      as="div"
      colorScheme="secondary"
      size="md"
      verticalMargin="micro"
    >
      описание
    </TypographyStyled>
    <PaddingWrapper
      verticalPadding="inner"
    >
      <span>
        foo bar baz
      </span>
    </PaddingWrapper>
  </ContentStyled>
  <ImageWrapperStyled>
    <ImgStyled
      alt="Заголовок"
    />
  </ImageWrapperStyled>
</ContainerStyled>
`;
